### PR TITLE
Some cleanup (followup to protobuf dedup)

### PR DESCRIPTION
**Pull request by adizere**
Closes: #306 
Closes: #513 

Fixes for these two problems:

- [x] MsgConnectionOpenAck includes a new field counterparty_connection_id. Some minimal changes were done to handle this as part of #292 but tests need to be added. Also, the test infrastructure here is very hard to follow, maybe some cleanup would be nice

- [x] `store_client_type` is never used and can be confusing (based on discussion with @CharlyCst) -- #513 

Deferred issues that will be handled later: 

- [ ] ~~tendermint ClientState has new fields proof_specs that is currently ignored.~~ -> will be handled together in #469 (because it depends on re-enabling client tests)

______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [x] If applicable: Unit tests written, added test to CI.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by adizere** [comment]
@cezarad and @vitorenesduarte if you have a chance to look over this, any thoughts would be appreciated!

File: modules/src/ics03_connection/handler/conn_open_ack.rs [hunk 1]
```diff
@@ -257,6 +262,14 @@ mod tests {
...
                         e,
                     );
+                    // Verify that the error kind matches
+                    if let Some(expected_kind) = test.error_kind {
+                        assert_eq!(
+                            &expected_kind,
+                            e.kind(),
+                            "conn_open_ack: expected error kind mismatches thrown error kind"
+                        )
+                    }
```
Comment (romac) [inline modules/src/ics03_connection/handler/conn_open_ack.rs:272]: 👍

**Review by romac — APPROVED** [review]

**Review by ancazamfir — APPROVED** [review]
Looks good! Thanks Adi!
